src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Conformance/ConformanceBuilderTests.cs

```diff
@@ -93,6 +93,33 @@ public void GivenAConformanceBuilder_WhenAddingDefaultInteractions_ThenParameter
             Assert.False(noParameters);
         }
 
+        [Fact]
+        public void GivenAConformanceBuilder_WhenAddingRestSearchParam_ThenTypeSearchParamIsAdded()
+        {
+            _builder.AddRestSearchParams();
+
+            ITypedElement statement = _builder.Build();
+
+            object typeDefinition = statement.Scalar($"CapabilityStatement.rest.searchParam.where(name = '_type').definition");
+
+            Assert.Equal("http://hl7.org/fhir/SearchParameter/type", typeDefinition.ToString());
+        }
+
+        [Fact]
+        public void GivenAConformanceBuilder_WhenAddingResourceSearchParam_ThenTypeSearchParamIsNotAddedUnderResource()
+        {
+            _searchParameterDefinitionManager.GetSearchParameters("Account")
+               .Returns(new[] { new SearchParameterInfo("_type", SearchParamType.Token, description: "description"), });
+
+            _builder.AddDefaultSearchParameters();
+
+            ITypedElement statement = _builder.Build();
+
+            object typeDocumentation = statement.Scalar($"{ResourceQuery("Account")}.searchParam.where(name = '_type').documentation");
```

Comment:
Would this work if we just had `statement.Scalar($"{ResourceQuery("Account")}.searchParam.where(name = '_type')")`? I ask because if for some reason we one day don't add documentation this test would no longer catch if `_type` was added back.

Reply:
statement.Scalar($"{ResourceQuery("Account")}.searchParam.where(name = '_type')") fetches null regardless object is present or not. To confirm if _type is added back without documentation, we can do like below
statement.Scalar($"{ResourceQuery("Account")}.searchParam.where(name = '_type').name")